### PR TITLE
feat(redis): Permit reading credentials from a volume mount

### PR DESCRIPTION
**Pull request by olivergondza**
Can be provided via `REDIS_CREDS_DIR_PATH`. Mirrors https://github.com/argoproj/argo-cd/pull/24597/.

**What does this PR do / why we need it**:

It does not pass sensitive data over env vars / cmdline arguments.

**Which issue(s) this PR fixes**:

https://issues.redhat.com/browse/GITOPS-9070

**How to test changes / Special notes to the reviewer**:

**Checklist**

* [x] Documentation update is required by this PR (and has been updated) OR no documentation update is required.

## Summary by CodeRabbit

* **New Features**
  * Load Redis credentials from a mounted directory via new CLI flag --redis-creds-dir-path (reads auth and optional auth_username); using the directory together with explicit username/password is disallowed. Default behavior unchanged when flag is not used.

* **Documentation**
  * Configuration reference updated with file-based Redis credential option and REDIS_CREDS_DIR_PATH env var.

* **Tests**
  * Added tests covering directory-based credential loading, edge cases, and error handling.

### CODE REVIEW

File: cmd/argocd-agent/agent.go [hunk 1]
```diff
@@ -427,3 +431,34 @@ func loadCreds(path string) (auth.Credentials, error) {
...
+			usernameData = []byte{}
+		}
+		redisUsername = strings.TrimSpace(string(usernameData))
+
+		passwordFile := filepath.Join(redisCredsDirPath, "auth")
+		passwordData, err := os.ReadFile(passwordFile)
+		if err != nil {
+			cmdutil.Fatal("Failed to read Redis password from %s: %v", passwordFile, err)
+		}
+		redisPassword = strings.TrimSpace(string(passwordData))
```
[the remark this answers is not in the document]
Comment (olivergondza) [reply cmd/argocd-agent/agent.go]: Redis creds cannot contain whitespace to the best of my knowledge. Either way, dangling whitespace is more likely to create problems than their removal. Disregard.

File: docs/configuration/reference/principal.md [hunk 2]
```diff
@@ -525,6 +525,33 @@ Redis server hostname and port.
...
+| **CLI Flag** | `--redis-creds-dir-path` |
+| **Environment Variable** | `REDIS_CREDS_DIR_PATH` |
+| **ConfigMap Entry** | N/A |
+| **Type** | String |
+| **Default** | `""` |
+
+The directory with `auth` file for Redis password.
+In kubernetes, this is intended to read a Secret mounted as a directory.
+
+Cannot be used together with `--redis-password`, or its respective environment variables.
```
[the remark this answers is not in the document]
Comment (olivergondza) [reply docs/configuration/reference/principal.md:541]: Disregard, principal does not read the username, apparently.

File: cmd/argocd-agent/agent.go [hunk 3]
```diff
@@ -427,3 +437,33 @@ func loadCreds(path string) (auth.Credentials, error) {
...
+		if err != nil {
+			// The file is optional, use an empty username if missing
+			if !os.IsNotExist(err) {
+				return "", "", fmt.Errorf("failed reading username from '%s': %v", usernameFile, err)
+			}
+			usernameData = []byte{}
+		}
+		username = strings.TrimSpace(string(usernameData))
+
+		passwordFile := filepath.Join(credsDirPath, "auth")
```
Comment (akhilnittala) [inline cmd/argocd-agent/agent.go:460]: what if passwordData is empty ?

File: cmd/argocd-agent/agent.go [hunk 4]
```diff
@@ -427,3 +437,33 @@ func loadCreds(path string) (auth.Credentials, error) {
...
+		passwordData, err := os.ReadFile(passwordFile)
+		if err != nil {
+			return "", "", fmt.Errorf("failed reading password from '%s': %v", passwordFile, err)
+		}
+		password = strings.TrimSpace(string(passwordData))
+	}
+
+	return username, password, nil
```
Comment (akhilnittala) [inline cmd/argocd-agent/agent.go:468]: if password is empty, it will return empty, so i feel it is better to add some validation above.

File: cmd/argocd-agent/agent_test.go [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_redisCreds(t *testing.T) {
```
Comment (akhilnittala) [inline cmd/argocd-agent/agent_test.go:26]: we can add data empty usecase here in the file

File: cmd/argocd-agent/agent.go (hunk 4 above)
Comment (olivergondza) [reply cmd/argocd-agent/agent.go:468]: Correct. It will be an empty string.

Note the `principal.WithRedis` and `agent.WithRedisPassword` handles empty string already, that is how missing option/envvar have manifests before.

File: cmd/argocd-agent/agent_test.go (hunk 5 above)
Comment (olivergondza) [reply cmd/argocd-agent/agent_test.go:26]: Added

File: cmd/argocd-agent/agent.go (hunk 3 above)
Comment (olivergondza) [reply cmd/argocd-agent/agent.go:460]: See my reply in comment bellow

File: cmd/argocd-agent/root_test.go [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+      --tls-cert string                               Use TLS certificate from path
+      --tls-ciphersuites strings                      Comma-separated list of TLS cipher suites to use. Use 'list' to show available cipher suites and exit
+      --tls-key string                                Use TLS private key from path
+      --tls-max-version string                        Maximum TLS version to accept (tls1.1, tls1.2, tls1.3)
+      --tls-min-version string                        Minimum TLS version to accept (tls1.1, tls1.2, tls1.3). Default: tls1.3 (default "tls1.3")
+      --tls-secret-name string                        Secret name of TLS certificate and key (default "argocd-agent-principal-tls")
+`
+)
+
+func TestNewRootCommand(t *testing.T) {
```
Comment (jannfis) [inline cmd/argocd-agent/root_test.go]: I think this test will be a PITA to maintain and it doesn't provide much value, to be honest. Please remove this test.

Comment (olivergondza) [reply cmd/argocd-agent/root_test.go]: :shrug: did not find a better way to test...
